pkg/infra/simulation/simulation.go

```diff
@@ -0,0 +1,108 @@
+package simulation
+
+import (
+	"log"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/agents"
+	agent1_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent1"
+	agent1_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent2"
+	agent2_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team2/agent1"
+	agent2_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team2/agent2"
+	agent3_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team3/agent1"
+	agent3_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team3/agent2"
+	agent4_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team4/agent1"
+	agent4_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team4/agent2"
+	agent5_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team5/agent1"
+	agent5_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team5/agent2"
+	agent6_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team6/agent1"
+	agent6_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team6/agent2"
+	agent7_1 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team7/agent1"
+	agent7_2 "github.com/SOMAS2021/SOMAS2021/pkg/agents/team7/agent2"
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra/tower"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/abm"
+	"github.com/google/uuid"
+)
+
+type SimEnv struct {
+	FoodOnPlatform  float64
+	AgentCount      []int
+	AgentHP         int
+	Iterations      int
+	reshufflePeriod int
+}
+
+func New(foodOnPlat float64, agentCount []int, agentHP, iterations, reshufflePeriod int) *SimEnv {
+
+	s := &SimEnv{
+		FoodOnPlatform:  foodOnPlat,
+		AgentCount:      agentCount,
+		AgentHP:         agentHP,
+		Iterations:      iterations,
+		reshufflePeriod: reshufflePeriod,
+	}
+	// can do other inits here
+	return s
+}
+
+type AgentNewFunc func(base *agents.Base) (abm.Agent, error)
+
+func (sE *SimEnv) Simulate() {
+	a := abm.New()
+
+	totalAgents := sum(sE.AgentCount)
+	t := tower.New(sE.FoodOnPlatform, 1, totalAgents, 1, 20, sE.reshufflePeriod)
+	a.SetWorld(t)
+
+	agentIndex := 1
+	for i := 0; i < len(sE.AgentCount); i++ {
+		for j := 0; j < sE.AgentCount[i]; j++ {
+			sE.createNewAgent(a, t, i, agentIndex)
+			agentIndex++
+		}
+	}
+	a.LimitIterations(sE.Iterations)
+	sE.simulationLoop(a, t)
+}
+
+func (sE *SimEnv) simulationLoop(a *abm.ABM, t *tower.Tower) {
+	// TODO: abstract with an extra function
+	for i := 1; i <= sE.Iterations; i++ {
+		missingAgentsMap := t.UpdateMissingAgents()
```

Comment:
I feel like this bit should probably be done elsewhere in a separate function called 'replace_dead_Agents' or smth

Reply:
See #20. Will be resolved post-MVP release.